lib/mocktail/imitates_type/makes_double/declares_dry_class.rb

```diff
@@ -42,21 +44,29 @@ def define_double_methods!(dry_class, type, instance_methods)
       handles_dry_call = @handles_dry_call
       instance_methods.each do |method|
         dry_class.undef_method(method) if dry_class.method_defined?(method)
+        parameters = type.instance_method(method).parameters
+        signature = @transforms_params.transform(Call.new, params: parameters)
+        method_signature = @builds_method_signature.build(signature)
 
-        dry_class.define_method method, ->(*args, **kwargs, &block) {
-          Debug.guard_against_mocktail_accidentally_calling_mocks_if_debugging!
-          handles_dry_call.handle(Call.new(
-            singleton: false,
-            double: self,
-            original_type: type,
-            dry_type: dry_class,
-            method: method,
-            original_method: type.instance_method(method),
-            args: args,
-            kwargs: kwargs,
-            block: block
-          ))
-        }
+        # TODO: This is failing because none of self, type, dry_class, method,
+        # etc are in scope for the dry_class.
+        dry_class.define_method method,
+          eval(<<-RUBY, binding, __FILE__, __LINE__ + 1)
+            ->#{method_signature} do
+              Debug.guard_against_mocktail_accidentally_calling_mocks_if_debugging!
+              handles_dry_call.handle(Call.new(
+                singleton: false,
+                double: self,
+                original_type: type,
+                dry_type: dry_class,
+                method: method,
+                original_method: type.instance_method(method),
+                args: args,
+                kwargs: kwargs,
```

Comment:
To get these I wonder if we can use BuildMethodSignature to get an array of symbols of the args and the kwargs, then use that to map an array of the values?

```ruby
builds_method_signature.args #=> [:positional1, :positional2]
build_method_signature.args.map { send(_1 } #=> ["value1", "value2"]
```

Reply:
Seems like a starting point.